### PR TITLE
Feature/add forms

### PR DESCRIPTION
**Pull request by Mhamad-Raad**
# Form
## In this branch I have done the followings:
- Added Form for like controller.
- Added Form for comment controller.
- Added Forms for Posts controller.
### Done 🎉 🔥 💯

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class ApplicationController < ActionController::Base
+  def current_user
+    User.last
+  end
```
Comment (codepantha) [inline app/controllers/application_controller.rb]: - [x] The `current_user` method should return the first user from the database according to the project requirement. Please implement this change 👍

File: config/routes.rb [hunk 2]
```diff
@@ -9,4 +9,16 @@
...
+
+  resources :posts do
+    resources :comments
+  end
+  post '/posts/new', to: 'posts#create' #create_new_post
+
+  resources :posts do 
+    resources :likes
+  end
+
```
Comment (codepantha) [inline config/routes.rb]: - [x] A lot of your routes are repeated. Apart from being confusing, it is not the best practice. Please clean up your routes, especially the ones related to `posts`. Also remember to only expose methods that you would be using.
I will give you a hint 👇 
```ruby
resources :users, only: [:index, :show] do
  resources :posts, only: [:index, :show, :new, :create] do
    resources :comments, only: [:new, :create]
    resources :likes, only: [:create]
  end
end

```

File: app/views/comments/new.html.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+<h2>comment added<h1>
```
Comment (codepantha) [inline app/views/comments/new.html.erb]: - [x] Please create a form here that accepts input for creating comments as per project requirements 👍 and ensure that it is rendered when a user visits the route below
```ruby
GET /posts/:post_id/comments/new(.:format) comments#new
```

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -3,14 +3,26 @@
...
-
+  <section>
+    <h2>ADD a comment</h2>
+      <%= form_with model: comment, local: true, url:  post_comments_path(@post), scope: @comment, class: "my-form" do |form| %>
+        <%= form.text_area :text, placeholder: "Write your comment here", class: "my-text-area" %>
+        <%= form.hidden_field :post_id, value: @post.id %>
+        <%= form.hidden_field :author_id, value: @user.id %>
+      <%= form.submit class: "my-submit-button"%>
+  <% end %>
+  </section>
```
Comment (codepantha) [inline app/views/posts/show.html.erb]: - [ ] You have a nice approach of having the comment form on the `post#show` page, however, the project requirement is that the comment should have it's own view and this form should be rendered in the `views/comments/new.html.erb` file
Please implement the following routes for the comments 👇 
```ruby
   POST   /posts/:post_id/comments(.:format)     comments#create
 GET   /posts/:post_id/comments/new(.:format)     comments#new
```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Mhamad-Raad 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Used GitFlow
✔️ Added forms to your app 📌 
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[# STATUS: APPROVED :green_circle: :clap: 

Hello  @Mhamad-Raad           , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: aa
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Descriptive P.R









 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]

**Pull request merged**